locust/web.py

```diff
@@ -477,4 +514,59 @@ def update_template_args(self):
             "stats_history_enabled": options and options.stats_history_enabled,
             "tasks": dumps({}),
             "extra_options": extra_options,
+            "show_locustfiles": self.show_locustfiles,
         }
+
+    def update_environment(self, locustfile, user_classes, shape_class, events, parsed_options):
+        options = parsed_options
+        environment = self.environment
+        runner = environment.runner
+
+        user_classes = list(user_classes.values())
+
+        # update properties
+        environment.__init__(
```

Comment:
Oh.. this looks a little shady to me. Re-running init? Why not just create a new object?

Reply:
Re-running `__init__` is not necessary, now that you mention it. The purpose of this was to just update the environment. I can just update each property individually or create a new object like you said.

Reply:
Yea, just overwrite the properties. environment.__init__ doesnt do a whole lot of validation so it is pretty much the same. Making a new object is cleaner, but maybe someone has a reference to it and could become confused.